core/lib/spree/testing_support/authorization_helpers.rb

```diff
@@ -2,54 +2,51 @@ module Spree
   module TestingSupport
     module AuthorizationHelpers
       module CustomAbility
-        def build_ability(&block)
-          block ||= proc{ |u| can :manage, :all }
-          Class.new do
-            include CanCan::Ability
-            define_method(:initialize, block)
-          end
+        def stub_permissions!(permission_sets: [Spree::PermissionSets::SuperUser])
+          # Since we cannot reliably stub the user or ability for capybara specs,
+          # we change the permissions for the default user, and change them back in the after(:all).
+          before(:all) { Spree::RoleConfiguration.instance.assign_permissions "default", permission_sets }
+          after(:all)  { Spree::RoleConfiguration.instance.unassign_permissions "default", permission_sets }
+        end
+        def custom_authorization!(&block)
+          ActiveSupport::Deprecation.warn "custom_authorization! will be deprecated in the next release, please use stub_authorization!", caller
+          stub_authorization! { block }
```

Comment:
Instead of passing the block literally you should be able to use the shorter (and also AST wise less complex form):

```
stub_authorization!(&block)
```

Which makes it more clear the captured block is forwarded unmodified.